Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,366 @@
+                        all_pods_ready = False
+                        logger.info(f"Pod {pod_name} in {cso_namespace} is not ready. Phase: {phase}, Ready: {readiness_states}")
+            else:
+                logger.error(f"Error fetching pods in {cso_namespace}: {result.stderr}")
+                all_pods_ready = False
+
+        except subprocess.CalledProcessError as error:
+            logger.error(f"Error checking pods in {cso_namespace}: {error}")
+            all_pods_ready = False
+
+        if all_pods_ready:
+            logger.info("All CSO pods in 'cso-system' namespace are ready.")
+            return True
+
+        logger.info("Waiting for CSO pods in 'cso-system' namespace to become ready...")
+        time.sleep(interval)
+
+    raise TimeoutError(f"Timed out after {timeout} seconds waiting for CSO pods in 'cso-system' namespace to become ready.")
+
+
+def wait_for_workload_pods_ready(namespace="kube-system", timeout=600, kubeconfig_path=None, max_retries=3, delay=30):
+    """
+    Waits for all pods in a specific namespace on a workload Kubernetes cluster to become ready.
+
+    :param namespace: The Kubernetes namespace where pods are located (default is "kube-system").
+    :param timeout: The timeout in seconds to wait for pods to become ready (default is 600).
+    :param kubeconfig_path: Path to the kubeconfig file for the target Kubernetes cluster.
+    :raises RuntimeError: If pods are not ready within the specified timeout.
+    """
+    for attempt in range(max_retries):
+        try:
+            kubeconfig_option = f"--kubeconfig {kubeconfig_path}" if kubeconfig_path else ""
+            wait_pods_command = (
+                f"kubectl wait -n {namespace} --for=condition=Ready --timeout={timeout}s pod --all {kubeconfig_option}"
+            )
+
+            # Run the command
+            subprocess.run(wait_pods_command, shell=True, check=True)
+            logger.info(f"All pods in namespace {namespace} in the workload Kubernetes cluster are ready.")
+            return True
+
+        except subprocess.CalledProcessError as error:
+            logger.warning(f"Attempt {attempt+1}/{max_retries} failed with error: {error}. Retrying in {delay} seconds...")
+            time.sleep(delay)
+
+    raise RuntimeError(f"Error waiting for pods in namespace '{namespace}' to become ready.")
+
+
+def load_config(config_path):
+    """
+    Loads the configuration from a YAML file.
+    """
+    if not os.path.exists(config_path):
+        raise FileNotFoundError(f"Configuration file {config_path} not found.")
+
+    with open(config_path, 'r') as file:
+        config = yaml.safe_load(file) or {}
+        return config
+
+
+def setup_environment_variables(self):
+    # Cluster Stack Parameters
+    self.clouds_yaml_path = self.config.get('clouds_yaml_path', '~/.config/openstack/clouds.yaml')
+    self.cs_k8s_version = self.cluster_version
+    self.cs_name = self.config.get('cs_name', 'scs')
```

Comment:
If we really need to hard-code all these defaults (which I doubt because we deliver an example file, and this thing won't work without a config file), then the defaults should be put into a dict, for instance

```python
DEFAULTS = {
    'cs_name': 'scs',
    'cs_version': 'v1',
    ...
}
```

Then I would do

```python
for key, value in DEFAULTS.items():
    self.config.setdefault(key, value)   # will only set if not already present
```

Also, for the environment variables, I think the detour with the instance variables is unnecessary.

You can do just

```python
required_env = {key.upper(): value for key, value in self.config}
```

Of course, this is not quite correct yet, because some values have to be computed differently, and not all keys from the config would be used. For the first point, we can always amend `required_env`, and for the second, we can add a guard to the dict comprehension like so:

```python
required_env = {key.upper(): value for key, value in self.config if key in ENV_KEYS}
```

where `ENV_KEYS` would be a global variable of type `set`, such as

```python
# don't use uppercase here because this refers to config.yaml keys
ENV_KEYS = {'cs_name', 'cs_version', ...}
```